File: Ahorn/entities/BanditoRoller.jl
module BanditoRoller

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/BanditoRoller" BanditoRoller(x::Integer, y::Integer, bounceSpeed::Integer=200, detectionRange::Integer=200, health::Integer=2, rollSpeed::Integer=150)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(BanditoRoller)
)

function Ahorn.selection(entity::BanditoRoller)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BanditoRoller, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: Loenn/entities/banditoRoller.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local banditoRoller = {}

banditoRoller.name = "MaggyHelper/BanditoRoller"
banditoRoller.depth = -100
banditoRoller.fieldInformation = {
    health = {
        minimumValue = 1
    },
    rollSpeed = {
        minimumValue = 0
    },
    bounceSpeed = {
        minimumValue = 0
    },
    detectionRange = {
        minimumValue = 0
    }
}
banditoRoller.placements = {
    {
        name = "normal",
        data = {
            health = 2,
            rollSpeed = 150,
            bounceSpeed = 200,
            detectionRange = 200
        }
    }
}

function banditoRoller.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/bandito_roller/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function banditoRoller.selection(room, entity)
    return utils.rectangle(entity.x - 12, entity.y - 12, 24, 24)
end

return banditoRoller
